torchvision/prototype/transforms/utils.py

```diff
@@ -50,18 +56,25 @@ def query_spatial_size(flat_inputs: List[Any]) -> Tuple[int, int]:
     if not sizes:
         raise TypeError("No image, video, mask or bounding box was found in the sample")
     elif len(sizes) > 1:
+        # (This comment applies to the other query_() utils):
+        # Does this happen?
+        # What can go terribly wrong if we don't raise an error?
+        # Should we just document that this returns the size of the first inpt to avoid an error?
```

Comment:
> Does this happen?

Hopefully not. If it does, the input data is setup incorrectly. For example two images in the same sample of different shape, or an `bounding_box.spatial_size` attribute out of sync with the corresponding image.

> What can go terribly wrong if we don't raise an error?

The first problem is that we needed to decide which value we return. `sizes.pop()` is not deterministic, so that would be quite bad.

Even if we can decide on a strategy to return a value, best case scenario is that the transform that requested the size fails loudly, albeit with a less expressive error message, when trying to transform an item that doesn't match the returned value. Worst case, the transform silently passes on bad input data.

Reply:
> The first problem is that we needed to decide which value we return. sizes.pop() is not deterministic, so that would be quite bad.

If we decide to not check for duplicates we don't need to create a `set`, so we don't need to `pop()` and we can return the size of the first input which is deterministic.

My first instinct would be to be defensive and check for duplicates as done in the current code. I wonder however if there's value in applying the "define errors out of existence" principle here. As you mentioned, the main risk is that the transforms silently returns wrong results - although it's not clear to me yet in which specific scenario that would happen.

(I'm not advocating for either option at this point, I'm merely trying to figure out the tradeoffs)

Do we have a sense of the performance hit incurred by the duplicate checks? If this gets called on large-ish batches (256 samples) with multiple inputs (images, masks, bbox, etc.), does it become noticeable?

Reply:
> Do we have a sense of the performance hit incurred by the duplicate checks?

Nope. We only tried and benchmarked on our references and there we didn't notice anything.

Reply:
Let's keep it this way then, thanks